packages/matter.js/src/behavior/definitions/operational-credentials/OperationalCredentialsServer.ts

```diff
@@ -366,7 +366,13 @@ export class OperationalCredentialsServer extends OperationalCredentialsBehavior
             };
         }
 
-        await fabric.remove(this.session.id);
+        // That's a workaround to make sure that factory reset works correctly in case of removal of the last fabric
+        // TODO find better solution
```

Comment:
Looks like a bug I should look at?

Reply:
Yes, I add a todo in the project

Reply:
Ha, I found the solution :-) check the newest commits please ... I think thats it.

The factory was started with the fabric removal, but sessions are expired afterwards and might need some round trips to finish with ack and such. so now the factory reset is triggered when the last session gets removed after fabric removal.

I also adjusted the logic that we do not announce the device as commissionable on last session closure to just expire that again because of factory reset. Have a look please.